modules/imgcodecs/test/test_grfmt.cpp

```diff
@@ -111,9 +117,9 @@ TEST(Imgcodecs_imread, regression)
         ASSERT_TRUE(imread_compare(path, IMREAD_COLOR));
         ASSERT_TRUE(imread_compare(path, IMREAD_ANYDEPTH));
         ASSERT_TRUE(imread_compare(path, IMREAD_ANYCOLOR));
-        if (path.substr(path.length() - 3) != "hdr")
+        if (path.substr(path.length() - 3) != "hdr" && path.substr(path.length() - 3) != "dcm")
```

Comment:
I think I'm going bettwer with the 
`const string ext = path.substr(path.length() -3);`
It will only generate one extra line of code, whereas making a separate file list will reduce readability somewhat.